storage-narpc/src/main/java/com/ibm/crail/storage/tcp/TcpStorageRequest.java

```diff
@@ -0,0 +1,198 @@
+/*
+ * Crail: A Multi-tiered Distributed Direct Access File System
+ *
+ * Author: Patrick Stuedi <stu@zurich.ibm.com>
+ *
+ * Copyright (C) 2016, IBM Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.ibm.crail.storage.tcp;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+
+import com.ibm.crail.conf.CrailConstants;
+import com.ibm.narpc.NaRPCMessage;
+
+public class TcpStorageRequest implements NaRPCMessage {
+	public static final int CSIZE = 4 + Math.max(WriteRequest.CSIZE, ReadRequest.CSIZE);
+	
+	private int type;
+	private WriteRequest writeRequest;
+	private ReadRequest readRequest;
+	
+	public TcpStorageRequest(){
+		writeRequest = new WriteRequest();
+		readRequest = new ReadRequest();
+	}
+
+	public TcpStorageRequest(WriteRequest writeRequest) {
+		this.writeRequest = writeRequest;
+		this.type = TcpStorageProtocol.REQ_WRITE;
+	}
+
+	public TcpStorageRequest(ReadRequest readRequest) {
+		this.readRequest = readRequest;
+		this.type = TcpStorageProtocol.REQ_READ;
+	}
+
+	public int size() {
+		return CSIZE;
+	}
+	
+	public int type(){
+		return type;
+	}
+
+	@Override
+	public void update(ByteBuffer buffer) throws IOException {
+		type = buffer.getInt();
+		if (type == TcpStorageProtocol.REQ_WRITE){
+			writeRequest.update(buffer);
+		} else if (type == TcpStorageProtocol.REQ_READ){
+			readRequest.update(buffer);
+		}
+	}
+
+	@Override
+	public int write(ByteBuffer buffer) throws IOException {
+		buffer.putInt(type);
+		int written = 4;
+		if (type == TcpStorageProtocol.REQ_WRITE){
+			written += writeRequest.write(buffer);
+		} else if (type == TcpStorageProtocol.REQ_READ){
+			written += readRequest.write(buffer);
+		}
+		return written;
+	}
+	
+	public static class WriteRequest {
+		public static final int CSIZE = 4 + 8 + 4 + 4 + (int) CrailConstants.BLOCK_SIZE;
```

Comment:
Don't use constants here but KEY_SIZE, ADDRESS_SIZE etc. otherwise this is hard to parse and easy to make mistakes